### PR TITLE
propagate consensus proposal hash in indexer

### PR DESCRIPTION
**Pull request by hhalex**

### CODE REVIEW

File: src/consensus.rs [hunk 1]
```diff
@@ -852,15 +857,70 @@ impl Consensus {
         Ok(())
     }
 
+    pub fn genesis_block(
+        initial_validators: Vec<ValidatorPublicKey>,
+        genesis_txs: Vec<Transaction>,
+    ) -> SignedBlock {
```
[the remark this answers is not in the document]
Comment (BertrandD) [reply src/consensus.rs:863]: imho this function should be in genesis.rs

File: src/data_availability.rs [hunk 2]
```diff
@@ -473,17 +417,17 @@ impl DataAvailability {
         if self.blocks.last().is_some() {
             if self
                 .blocks
-                .get(block.parent_hash.clone())
+                .get(block.parent_hash())
                 .unwrap_or(None)
                 .is_none()
             {
-                debug!(
+                warn!(
```
Comment (BertrandD) [inline src/data_availability.rs:424]: I don't think this should be warn. A new node cathing up will log a warn for each block, this is a "normal" case when catching up

File: src/indexer/migrations/20241009154948_create_tables.sql [hunk 3]
```diff
@@ -1,11 +1,11 @@
 -- Add migration script here
 CREATE TABLE blocks (
-    hash TEXT PRIMARY KEY,          -- Corresponds to BlockHash 
-    parent_hash TEXT NOT NULL,      -- Parent block hash (BlockHash)
+    hash BYTEA PRIMARY KEY,          -- Corresponds to BlockHash
```
Comment (BertrandD) [inline src/indexer/migrations/20241009154948_create_tables.sql:3]: when doing debug, having a TEXT field is very much easier to query than a BYTEA, I highly recommend to keep a string there

**Review by BertrandD — CHANGES_REQUESTED** [review]

File: src/consensus.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (wraitii) [reply src/consensus.rs:863]: Found the answer -> we need the consensus proposal or we won't be able to get the hash.
I work around it in my version of this diff

**Comment by wraitii** [comment]
Done in #504

File: src/indexer/migrations/20241009154948_create_tables.sql (hunk 3 above)
Comment (hhalex) [reply src/indexer/migrations/20241009154948_create_tables.sql:3]: Not sure I agree... You can encode/decode it with `encode(column, "hex")` when doing debug

**Pull request closed without merging**